lib/std/Build.zig

```diff
@@ -1888,7 +1948,7 @@ pub fn anonymousDependency(
         '/', '\\' => byte.* = '.',
         else => continue,
     };
-    return dependencyInner(b, name, build_root, build_zig, &.{}, args);
+    return dependencyInner(b, name, build_root, build_zig, "anonymous", &.{}, args);
```

Comment:
You can ignore `"anonymous"` here, it's just a dummy value that makes it so that package hash comparisons always fail.
Removing `anonymousDependency` is an accepted proposal and I have a different PR up that removes it (#18590).